linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/ClassEmitter.scala

```diff
@@ -233,7 +236,19 @@ class ClassEmitter(coreSpec: CoreSpec) {
         ancestors.headOption.contains(className),
         s"The ancestors of ${className.nameString} do not start with itself: $ancestors"
       )
-      val strictAncestors = ancestors.tail
+      val strictAncestors0 = ancestors.tail
+
+      // If Class_superClass is used, and the class has a super class, move it first
```

Comment:
Just out of curiosity, am I correct that, `superClass` won't be on the head of `strictAncestors0` when the class implements any interfaces?

Reply:
Besides `ancestors.head`, there is no guarantee at all in what order ancestors show up in `ancestors.tail`. For all we know we could even have a grand-parent class come before the direct parent class.